### ISSUE TITLE
Getting openssl error (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', '', 'wrong signature type')])")))

### ISSUE THREAD

**Issue by RejoyMR** [openssl/openssl#18264]
toot@ubuntu-20:/opt/openssl/ssl# pfsso -r $account -u $email-address -g eu-central-1 -s 8 --tokenmode ask
SSO Password: 
HTTPSConnectionPool(host='authenticate.microfocus.net', port=443): Max retries exceeded with url: /nidp/app/login?id=MultiGenericAA&sid=0&option=credential (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', '', 'wrong signature type')])")))
Error logging you in to IT Corp AD.
    Please verify your credentials by logging in manually

**Comment by mattcaswell** [comment]
There can be a number of causes for this error, but the most likely is that the peer's signature algorithm is insufficiently strong. I believe that Ubuntu sets the default OpenSSL security level to 2 (the project itself sets the default to 1). You might try overriding it and see if it makes a difference, e.g. create an OpenSSL config file like this:

````
openssl_conf = default_conf

[ default_conf ]

ssl_conf = ssl_sect

[ssl_sect]

system_default = ssl_default_sect

[ssl_default_sect]
CipherString = DEFAULT:@SECLEVEL=1
````

Then set the `OPENSSL_CONF` environment variable to point at it and try your command again, e.g.

````
# OPENSSL_CONF=/path/to/my/openssl.cnf pfsso -r $account -u $email-address -g eu-central-1 -s 8 --tokenmode ask
````

If that doesn't work you might try lowering the security level even further to 0.

**Comment by RejoyMR** [comment]
I tried this above options but did not help. Any other solution. 

I am getting same error.. 

TTPSConnectionPool(host='authenticate.microfocus.net', port=443): Max retries exceeded with url: /nidp/app/login?id=MultiGenericAA&sid=0&option=credential (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', '', 'wrong signature type')])")))

**Comment by mattcaswell** [comment]
Getting a wireshark dump of the communication might give us some hints as to what is going on.

**Comment by nhorman** [comment]
Marking as inactive, to be closed when 3.4 dev cycle ends, barring further input
